### ISSUE TITLE
Model.save_weights() then Model.load_weights() gives different results.

### ISSUE THREAD

**Issue by saeedalahmari3** [fchollet/keras#8498]
I have trained a model and I used checkpoint to save the best model. Then I evaluated the model in validation set using model.evaluate_generator(), it gives 73%. When I loaded the model weights using model.load_weights() and then evaluate generator it gave me 80%.  Why is that happening? Any Idea?

**Comment by saeedalahmari3** [comment]
OK, I got the problem. in checkpoint I didn't set save_weights_only to True. This fixed the problem.
